---
title: "Week2: playing with annotations"
author: "Pierre-Luc"
output: html_document
---

```{r}
suppressPackageStartupMessages({
  library(AnnotationHub, lib.loc="../env") # to fetch genomes, annotations, etc.
  library(ensembldb, lib.loc="../env")     # to manipulate ensembl gene annotations
  library(GenomicRanges, lib.loc="../env") # data structure for genomic ranges
  library(backports, lib.loc="../env")
  library(epiwraps, lib.loc="../env")      # visualization
})
```

# Fetch the drosophila gene annotation

```{r}
ah <- AnnotationHub()
# if you're on renku, use ah <- AnnotationHub(cache="ahcache")
# to query the AH programmatically:
q <- query(ah, c("Drosophila", "ensembl"))
# we can check the objects that match:
q

ensdb <- ah[["AH95713"]]
ensdb
```


```{r}
g <- genes(ensdb)
head(g)
length(g)

table(g$gene_biotype)

pcg <- g[which(g$gene_biotype=="protein_coding")]
head(pcg)
length(pcg)

ex <- exons(ensdb)
head(ex)
ex <- exonsBy(ensdb)
head(ex)

numExonsPerTx <- lengths(ex)
head(numExonsPerTx)
hist(numExonsPerTx)
```

```{r}
# from the `ensembldb` package
# genes(ensdb)
# extract transcripts:
txs <- transcripts(ensdb)
txs
seqlevels(txs)
head(txs$tx_biotype)
head(txs[,"tx_biotype"])
table(seqnames(txs))
table(txs$tx_biotype)

# supportedFilters(), tx_biotype

exs <- exons(ensdb, column=c("tx_id"))
exs

# same as "exons()" but returning the exons split by transcripts
exsPerTx <- exonsBy(ensdb, column=c("tx_id","tx_biotype"))
# we limit ourselves to protein-coding transcripts:
exsPerTx <- exonsBy(ensdb, column=c("tx_id","tx_biotype"), 
                    filter=TxBiotypeFilter("protein_coding"))
exsPerTx

# what are the transcripts that have more than on exon / sub-element:
head(which(lengths(exsPerTx)>1))

exsPerTx[["FBtr0005088"]]
```

How many transcripts do protein-coding genes have?

```{r}
nbExonsPerPCtx <- lengths(exsPerTx)
hist(nbExonsPerPCtx)
```

Selecting transcripts from one chromosome:

```{r}
txs_2L <- txs[seqnames(txs)=="2L"] 
```

Example of overlapping GRanges objects:

```{r}
# we create a new GRanges
gr <- GRanges("2L", 
              IRanges(start=c(14615552, 14616303),
                      end=c(14615552+10, 14616303+20)))
gr
# how many transcripts of 2L overlap with my `gr` object?
table(overlapsAny(txs_2L, gr, minoverlap=1, type="any"))
# sum(countOverlaps(txs_2L, gr, minoverlap=1, type="any")) => slightly different result because some transcripts overlap more than once: txs_2L[countOverlaps(txs_2L, gr, minoverlap=1, type="any")==2]
nearest(gr, txs_2L)

# nearest selects the closest, and if several have a distance of 0, randomly reports one
# we can report all with:
nearest(gr, txs_2L, select="all")

# we could also store the corresponding nearest transcript name 
# in the object of our regions of interest:
gr$nearestTx <- txs_2L$tx_name[nearest(gr, txs_2L)]

```

