---
title: "Final Project"
author: "Francesco Biancchi and Lukas Haeuser"
date: "2024/07/02"
output: html_document
---

```{r}
suppressPackageStartupMessages({
  library(epiwraps)
  library(AnnotationHub)
  library(MotifDb)
  library(memes)
  library(universalmotif)
  library(ensembldb)
  library(ggplot2)
  library(motifmatchr)
  library(SummarizedExperiment)
  library(sechm)
  library(BiocParallel)
  library(chromVAR)
  library(limma)
})
```



```{r}
ah <- AnnotationHub::AnnotationHub()
```

# Plot enrichment heatmaps
(Code is from week07)

```{r}
# get CTCF motif
motif <- query(MotifDb, c("CTCF","Mus"))[[1]]
motif2 <- convert_motifs(motif, class="TFBSTools-PFMatrix")
genome <- ah[["AH68356"]]
# get the sequence for chr1:
chr1 <- rtracklayer::import(genome)["1"]

# find motif matches across chr19
moi <- motifmatchr::matchMotifs(motif2, chr1, out="positions", p.cutoff=1e-5)[[1]]
# convert to GRanges
moi <- as(setNames(moi,names(chr1)), "GRanges")
```

```{r, fig.width=8, fig.height=4}
tracks <- list.files(pattern="bw$")
names(tracks) <- gsub("\\.bw","",basename(tracks))

sm <- signal2Matrix(tracks, moi, w=5, extend=300)
plotEnrichedHeatmaps(sm, trim=0.95)
```

```{r}
nf <- getNormFactors(tracks, useSeqLevels="1", nwind=5000L)
sm <- renormalizeSignalMatrices(sm, scaleFactors = nf)

plotEnrichedHeatmaps(sm, trim=0.95, minRowVal = 12, colors = c("white","darkred"))
```

# Differential Analysis
(Code is from week08)

```{r}
# get motifs 
motifs <- query(MotifDb, c("HOCOMOCOv10", "Mmusculus"))
# convert to a format motifmatchr can use, and use the gene symbols as names
motifs <- do.call(TFBSTools::PWMatrixList, setNames(
           universalmotif::convert_motifs(motifs, class="TFBSTools-PWMatrix"),
           mcols(motifs)$geneSymbol))
motifs
```

```{r}
# get genome
genome <- ah[["AH68356"]]
```
```{r, fig.width=8, fig.height=4}
# get samples files
tracks <- list.files(pattern="bw$")
names(tracks) <- gsub("\\.bw","",basename(tracks))
tracks
```


```{r}
# get the peaks
peak <- rtracklayer::import(genome)["1"] ### WHAT DO WE NEED TO USE AS PEAK???
hist(width(peak))
# we need to resize the peaks so that they all have a comparable size
# usually data is paired, but note here
# if data is not paired but paired=true, function throws error
peak <- resize(peak, width=300, fix="center")

se <- chromVAR::getCounts(alignment_files = bams, peaks = peak, paired = FALSE)
se
```
